sdk/communication/Azure.Communication.Administration/src/Models/InternalPhoneNumberReservationOperation.cs

```diff
@@ -0,0 +1,135 @@
+        /// <param name="cancellationToken">The cancellation token to use.</param>
+        internal InternalPhoneNumberReservationOperation(
+            PhoneNumberAdministrationClient client,
+            string phoneNumberReservationId,
+            Response initialResponse,
+            IReadOnlyList<ReservationStatus> terminateStatuses,
+            CancellationToken cancellationToken)
+        {
+            Id = phoneNumberReservationId;
+            _value = null;
+            _rawResponse = initialResponse;
+            _finalRawResponse = null!;
+            _client = client;
+            _cancellationToken = cancellationToken;
+            _terminateStatuses = terminateStatuses;
+        }
+
+        /// <inheritdocs />
+        internal string Id { get; }
+
+        /// <inheritdocs />
+        internal PhoneNumberReservation Value => OperationHelpers.GetValue(ref _value);
+
+        /// <inheritdocs />
+        internal bool HasCompleted => _hasCompleted;
+
+        /// <inheritdocs />
+        internal bool HasValue => (_rawResponse != null || _finalRawResponse != null) && _value != null;
+
+        /// <inheritdocs />
+        internal Response GetRawResponse() => HasCompleted ? _finalRawResponse : _rawResponse;
+
+        /// <summary>
+        /// Check for the latest status of the operation.
+        /// </summary>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>The <see cref="Response"/> with the status update.</returns>
+        internal Response UpdateStatus(CancellationToken cancellationToken = default)
+            => UpdateStatusAsync(false, cancellationToken).EnsureCompleted();
+
+        /// <summary>
+        /// Check for the latest status of the operation.
+        /// </summary>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>The <see cref="Response"/> with the status update.</returns>
+        internal async ValueTask<Response> UpdateStatusAsync(CancellationToken cancellationToken = default)
+            => await UpdateStatusAsync(true, cancellationToken).ConfigureAwait(false);
+
+        /// <summary>
+        /// Check for the latest status of the operation.
+        /// </summary>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <param name="async" />
+        /// <returns>The <see cref="Response"/> with the status update.</returns>
+        private async ValueTask<Response> UpdateStatusAsync(bool async, CancellationToken cancellationToken)
+        {
+            if (HasCompleted)
```

Comment:
We could add a DiagnosticScope for the body of `UpdateStatusAsync` everywhere, not sure if we need to.

Reply:
We should add to all.